### PR TITLE
Type cast for prevention of integer overflow

### LINKED ISSUE #26423: Unable to put items with charges into cargo dimension, other items are ok
**Describe the bug**  
Moving items into a cargo dimension works, but moving items with charges like ammo/duct tape gets a message stating the destination is full.

**To Reproduce**  
1. Have a cargo dimension on a vehicle
2. Move an item in
3. Move some ammo in

**Expected behavior**  
Both should be able to work if there is enough space in the cargo dimension

**Screenshots**  
Before moving the gun:
beforemovegun

After moving the gun:
aftermovegun

Trying to move the ammo:
tryingtomoveammo

**Versions and configuration(please complete the following information):**
- First noticed on build 8080, downgraded to 8078 with the same issue. 8076 has no such issue. Suspect it has to do with PR Fix truncated per-charge volume (#26031) in build 8077.

### PR DESCRIPTION
**Pull request by Zetsukaze**
#### Summary
```SUMMARY: Bugfixes "Type cast for prevention of integer overflow with large volume containers like cargo containers."```

#### Purpose of change
Fixes #26423 - Unable to put items with charges into cargo dimension

#### Describe the solution
Cast the integers to int64_t

#### Describe alternatives you've considered
Switching the equation to have the division calculated first before the multiplication. However was advised that that creates inaccuracies, and that casting to int64_t would be better.

### CODE REVIEW

**Comment by kevingranade** [comment]
Fix looks good, but you need to wrap the line because it's over the line length limit now.

File: src/item.cpp [hunk 1]
```diff
@@ -535,7 +535,8 @@ long item::charges_per_volume( const units::volume &vol ) const
     if( type->volume == 0 ) {
         return INFINITE_CHARGES; // TODO: items should not have 0 volume at all!
     }
-    return count_by_charges() ? vol * type->stack_size / type->volume : vol / volume();
+    // Type cast for prevention of integer overflow with large volume containers like the cargo dimension
+    return count_by_charges() ? static_cast<int64_t>(units::to_milliliter(vol)) * static_cast<int64_t>(type->stack_size) / static_cast<int64_t>(units::to_milliliter(type->volume)) : vol / volume();
```
Comment (jbytheway) [inline src/item.cpp]: More complicated than it needs to be.  You can just cast `type->stack_size` and the rest will work itself out. No need to convert or cast the volumes.

**Comment by jbytheway** [comment]
Looks like this fix didn't work for the mingw cross-compiles (which is the platfrom where `long` is 32 bits).
Unfortunately the CI didn't run right on this PR, but you can see it in the failures on my other PR here:
https://travis-ci.org/CleverRaven/Cataclysm-DDA/jobs/448522718

**Comment by Zetsukaze** [comment]
Hmm, how should we go about resolving this?

**Comment by jbytheway** [comment]
Suggest a quick & dirty fix such as commenting out the problematic test just to get the tests passing on master once more, and then figure out what's going on with the mingw builds in slower time.

**Pull request merged**